### DISCUSSION TITLE
How do I use my api gateways path/ and not /

### DISCUSSION THREAD

**Discussion by bclements** [go-chi/chi#905] [Q&A]
Hi,

We are using AWS api gateway that points to our chi service. 

apigw.domain.com/blogs -> blogs.domain.com

When we go to any endpoint hosted by our service using the apigw, example apigw.domain.com/blogs/new we get a not found error and the path dev tools says can be found is apigw.domain.com/new ....Looks ilke the chi router is not respecting the original path.

Is there a way to configure chi to respect the path used by the apigw?

**Comment by adilburaksen** [comment]
AWS API Gateway strips the routing prefix before forwarding to your backend service. When you configure a route `apigw.domain.com/blogs → blogs.domain.com`, API Gateway by default forwards the request to your service with the `/blogs` prefix removed — so `apigw.domain.com/blogs/new` arrives at your chi router as `/new`, not `/blogs/new`.

**Fix 1: Register routes without the prefix in chi**

This is the simplest approach. Since API Gateway strips `/blogs`, your chi router should just register `/new`:

```go
r.Get("/new", myHandler)
// not r.Get("/blogs/new", ...)
```

This is the standard pattern for API Gateway integrations.

**Fix 2: Configure API Gateway to pass the full path**

In AWS API Gateway (HTTP API), you can use a `$request.path` override in the integration settings to forward the full original path including the prefix. In REST API, use proxy integration with `{proxy+}` and set the integration URL to your backend with `/$request.path`.

**Fix 3: Use `StripPrefix` in chi (if you want both to work)**

If you're testing locally without a gateway and need the prefix, mount with `StripPrefix`:

```go
r.Mount("/blogs", http.StripPrefix("/blogs", blogRouter))
```

But for production behind API Gateway, option 1 is cleanest — let API Gateway handle the routing and keep your chi routes clean.
